storm-core/src/clj/org/apache/storm/daemon/worker.clj

```diff
@@ -487,11 +486,12 @@
             )))))
 
 (defn register-callbacks [worker]
-  (log-message "Registering IConnectionCallbacks for " (:assignment-id worker) ":" (:port worker))
-  (msg-loader/register-callback (:transfer-local-fn worker)
-                                (:receiver worker)
-                                (:storm-conf worker)
-                                (worker-context worker)))
+  (let [transfer-local-fn (:transfer-local-fn worker)
+        ^IConnection socket (:receiver worker)]
+    (log-message "Registering IConnectionCallbacks for " (:assignment-id worker) ":" (:port worker))
+    (.registerRecv socket (DeserializingConnectionCallback. (:storm-conf worker)
```

Comment:
Could we just inline `DeserializingConnectionCallback`? It's only used here and it'd be only a couple lines of Clojure. We're going to have to get rid of Java code that references Clojure like `DeserializingConnectionCallback` does anyway.